pkg/ccl/backupccl/backup_test.go

```diff
@@ -5674,7 +5676,280 @@ func TestProtectedTimestampsDuringBackup(t *testing.T) {
 		})
 		require.NoError(t, g.Wait())
 	}
+}
+
+// TestSpanSelectionDuringBackup tests the method spansForAllTableIndexes which
+// is used to resolve the spans which will be backed up, and spans for which
+// protected ts records will be created.
+func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	skip.UnderStressRace(t,
```

Comment:
Calling out that I've skipped this test on stressrace since all the tests use and mutate `actualResolvedSpans`. The only solution I could think of involved creating a test server for each subtest. Is this worth it or is it sufficient to run the test under normal conditions?